distributed/tests/test_worker_client.py

```diff
@@ -217,3 +217,31 @@ def func(x):
     future = c.submit(func, 10)
     result = yield future
     assert result == 11
+
+
+def test_secede_without_stealing_issue_1262():
+    """
+    Tests that seceding works with the Stealing extension disabled
+    https://github.com/dask/distributed/issues/1262
+    """
+
+    # turn off all extensions
+    extensions = []
+
+    # run the loop as an inner function so all workers are closed
+    # and exceptions can be examined
+    @gen_cluster(client=True, scheduler_kwargs={'extensions': extensions})
+    def secede_test(c, s, a, b):
+        def func(x):
+            with worker_client() as wc:
+                y = wc.submit(lambda: 1 + x)
+                return wc.gather(y)
```

Comment:
Looks like this isn't Python 2 compatible.  You'll have to use `raise gen.Return(...)`